File: 프로그래머스/1/12926. 시저 암호/시저 암호.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s, int n) {
    string answer = "";
        for (int i = 0; i < s.size(); i++)
    {
        if (s[i] >= 'A' && s[i] <= 'Z')
        {
            char shifted = (s[i] - 'A' + n) % 26 + 'A';
            answer += shifted;
        }
        else if (s[i] >= 'a' && s[i] <= 'z')
        {
            char shifted = (s[i] - 'a' + n) % 26 + 'a';
            answer += shifted;
        }
        else
        {
            answer += s[i];
        }
    }
    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.cs
public class Solution {
    public string solution(string s, int n) {
        string answer = "";
        for(int i=0;i<s.Length;i++)
        {
            if (s[i] == ' ')
            {
                answer += ' ';
                continue;
            }
            if (char.IsUpper(s[i]))
            {
                if (s[i] + n > 'Z')
                {
                    answer += (char)(s[i] + n - 26);
                    continue;
                }
            }
            else if (char.IsLower(s[i]))
            {
                if (s[i] + n > 'z')
                {
                    answer += (char)(s[i] + n - 26);
                    continue;
                }
            }
            answer += (char)(s[i] + n);
        }
        return answer;
    }
}